trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/generate_propeller_wake_distribution.py

```diff
@@ -0,0 +1,209 @@
+        Y_pts  = prop.origin[i][1] + (y_pts*wake_contraction)*azi_y  + sy_inf    
+    
+        z0_pts = np.tile(np.atleast_2d(r),(B,1))
+        z_pts  = np.repeat(np.repeat(z0_pts[np.newaxis,:,  :], nts, axis=0)[ np.newaxis, : ,:, :,], m, axis=0)
+        Z_pts  = prop.origin[i][2] + (z_pts*wake_contraction)*azi_z + sz_inf     
+        
+        # Store points  
+        # ( control point,  prop ,  time step , blade number , location on blade )
+        if (prop.rotation != None) and (prop.rotation[i] == -1):  
+            WD_XA1[:,i,:,:,:] =  X_pts[: , :-1 , : , :-1 ]
+            WD_YA1[:,i,:,:,:] =  Y_pts[: , :-1 , : , :-1 ]
+            WD_ZA1[:,i,:,:,:] =  Z_pts[: , :-1 , : , :-1 ]
+            WD_XA2[:,i,:,:,:] =  X_pts[: ,  1: , : , :-1 ]
+            WD_YA2[:,i,:,:,:] =  Y_pts[: ,  1: , : , :-1 ]
+            WD_ZA2[:,i,:,:,:] =  Z_pts[: ,  1: , : , :-1 ]
+            WD_XB1[:,i,:,:,:] =  X_pts[: , :-1 , : , 1:  ]
+            WD_YB1[:,i,:,:,:] =  Y_pts[: , :-1 , : , 1:  ]
+            WD_ZB1[:,i,:,:,:] =  Z_pts[: , :-1 , : , 1:  ]
+            WD_XB2[:,i,:,:,:] =  X_pts[: ,  1: , : , 1:  ]
+            WD_YB2[:,i,:,:,:] =  Y_pts[: ,  1: , : , 1:  ]
+            WD_ZB2[:,i,:,:,:] =  Z_pts[: ,  1: , : , 1:  ] 
+        else: 
+            WD_XA1[:,i,:,:,:] = X_pts[: , :-1 , : , 1:  ]
+            WD_YA1[:,i,:,:,:] = Y_pts[: , :-1 , : , 1:  ]
+            WD_ZA1[:,i,:,:,:] = Z_pts[: , :-1 , : , 1:  ]
+            WD_XA2[:,i,:,:,:] = X_pts[: ,  1: , : , 1:  ]
+            WD_YA2[:,i,:,:,:] = Y_pts[: ,  1: , : , 1:  ]
+            WD_ZA2[:,i,:,:,:] = Z_pts[: ,  1: , : , 1:  ] 
+            WD_XB1[:,i,:,:,:] = X_pts[: , :-1 , : , :-1 ]
+            WD_YB1[:,i,:,:,:] = Y_pts[: , :-1 , : , :-1 ]
+            WD_ZB1[:,i,:,:,:] = Z_pts[: , :-1 , : , :-1 ]
+            WD_XB2[:,i,:,:,:] = X_pts[: ,  1: , : , :-1 ]
+            WD_YB2[:,i,:,:,:] = Y_pts[: ,  1: , : , :-1 ]
+            WD_ZB2[:,i,:,:,:] = Z_pts[: ,  1: , : , :-1 ]
+        
+        WD_GAMMA[:,i,:,:,:] = Gamma 
+        
+        # store points for plotting 
+        VD.Wake.XA1[i,:,:,:] =  X_pts[0 , :-1 , : , :-1 ]
+        VD.Wake.YA1[i,:,:,:] =  Y_pts[0 , :-1 , : , :-1 ]
+        VD.Wake.ZA1[i,:,:,:] =  Z_pts[0 , :-1 , : , :-1 ]
+        VD.Wake.XA2[i,:,:,:] =  X_pts[0 ,  1: , : , :-1 ]
+        VD.Wake.YA2[i,:,:,:] =  Y_pts[0 ,  1: , : , :-1 ]
+        VD.Wake.ZA2[i,:,:,:] =  Z_pts[0 ,  1: , : , :-1 ]
+        VD.Wake.XB1[i,:,:,:] =  X_pts[0 , :-1 , : , 1:  ]
+        VD.Wake.YB1[i,:,:,:] =  Y_pts[0 , :-1 , : , 1:  ]
+        VD.Wake.ZB1[i,:,:,:] =  Z_pts[0 , :-1 , : , 1:  ]
+        VD.Wake.XB2[i,:,:,:] =  X_pts[0 ,  1: , : , 1:  ]
+        VD.Wake.YB2[i,:,:,:] =  Y_pts[0 ,  1: , : , 1:  ]
+        VD.Wake.ZB2[i,:,:,:] =  Z_pts[0 ,  1: , : , 1:  ]  
+          
+    
+    # Compress Data into 1D Arrays  
+    WD.XA1  =  np.reshape(np.reshape(np.reshape(WD_XA1,(m,num_prop,(nts-1),B*n)),(m,num_prop,(nts-1)*B*n)),(m,num_prop*(nts-1)*B*n))
```

Comment:
same here, make some tuples. It will be easier to read and computationally faster

Reply:
tuples made.